hedera-mirror-rest/controllers/allowanceController.js

```diff
@@ -0,0 +1,151 @@
+  /**
+   * Extracts SQL where conditions, params, order, and limit
+   *
+   * @param {[]} filters parsed and validated filters
+   * @param {Number} accountId parsed accountId from path
+   * @param {Number} startPosition param index start position
+   */
+  extractCryptoAllowancesQuery = (filters, accountId, startPosition = 1) => {
+    let limit = defaultLimit;
+    let order = constants.orderFilterValues.DESC;
+    const conditions = [`${CryptoAllowance.OWNER} = $${startPosition}`];
+    const params = [accountId];
+    const spenderInValues = [];
+
+    for (const filter of filters) {
+      if (_.isNil(filter)) {
+        continue;
+      }
+
+      switch (filter.key) {
+        case constants.filterKeys.SPENDER_ID:
+          if (!utils.isRegexMatch(constants.queryParamOperatorPatterns.eq, filter.operator)) {
+            throw new InvalidArgumentError(
+              `Only equals (eq) comparison operator is supported for ${constants.filterKeys.SPENDER_ID}`
+            );
+          }
+          this.updateConditionsAndParamsWithInValues(
+            filter,
+            spenderInValues,
+            params,
+            conditions,
+            CryptoAllowance.SPENDER,
+            startPosition + conditions.length
+          );
+          break;
+        case constants.filterKeys.LIMIT:
+          limit = filter.value;
+          break;
+        case constants.filterKeys.ORDER:
+          order = filter.value;
+          break;
+        default:
+          break;
+      }
+    }
+
+    this.updateQueryFiltersWithInValues(params, conditions, spenderInValues, CryptoAllowance.SPENDER);
+
+    return {
+      conditions,
+      params,
+      order,
+      limit,
+    };
+  };
+
+  /**
+   * Handler function for /accounts/:accountAliasOrAccountId/allowances/crypto API
+   * @param {Request} req HTTP request object
+   * @param {Response} res HTTP response object
+   * @returns {Promise<void>}
+   */
+  getAccountCryptoAllowances = async (req, res) => {
+    // extract filters from query param
+    let accountId = null;
+    try {
+      accountId = await EntityService.getEncodedIdOfValidatedEntityId(req.params.accountAliasOrAccountId);
+    } catch (err) {
+      if (err instanceof InvalidArgumentError) {
+        throw InvalidArgumentError.forParams(constants.filterKeys.ACCOUNT_ID_OR_ALIAS);
+      }
+
+      // rethrow any other error
+      throw err;
+    }
+
+    // extract filters from query param
+    const filters = utils.buildAndValidateFilters(req.query);
+
+    const {conditions, params, order, limit} = this.extractCryptoAllowancesQuery(filters, accountId);
+    const allowances = await CryptoAllowanceService.getAccountCrytoAllowances(conditions, params, order, limit);
+
+    const response = {
+      allowances: allowances.map((allowance) => new CryptoAllowanceViewModel(allowance)),
+      links: {
+        next: null,
+      },
+    };
+
+    if (!_.isEmpty(response.allowances) && response.allowances.length === limit) {
```

Comment:
nit, `response.allowances.length === limit` implies `response.allowances` isn't an empty array

```suggestion
    if (response.allowances.length === limit) {
```

Reply:
Updated